pyanaconda/bootloader.py

```diff
@@ -1456,6 +1453,18 @@ class GRUB2(GRUB):
     # XXX we probably need special handling for raid stage1 w/ gpt disklabel
     #     since it's unlikely there'll be a bios boot partition on each disk
 
+    # this is a property because the state of flags.blscfg changes
+    # during anaconda run, so it can't just be a class attribute
```

Comment:
The flag shouldn't change during the Anaconda run. It should be set before this module is imported. If you have in mind the pull request https://github.com/rhinstaller/anaconda/pull/1708, that code is run after the package installation, so this property shouldn't be used after that (at least I think so).

Reply:
I actually tested it, and nope. If I just put this conditional in the setting of the class variable, it doesn't work. You can try it for yourself easily enough - just generate an updates.img with that version of the change, and try an install of https://kojipkgs.fedoraproject.org/compose/rawhide/Fedora-Rawhide-20181130.n.0/compose/Server/x86_64/iso/Fedora-Server-dvd-x86_64-Rawhide-20181130.n.0.iso with `inst.noblscfg`. It doesn't install `grubby-deprecated`. With this version of the change, it does.

I agree it's unexpected, but that's the behaviour I experienced in testing. I can only assume that somehow `bootloader.py` gets imported *before* the code in `anaconda.py` that sets `flags.blscfg` is run.

Reply:
Well, I suppose it's possible that's to do with the updates mechanism, thinking about it (though I thought updates were applied during initramfs phase these days). I'll test it with a new image rather than just an updates.img.

Reply:
I am ok with the property, but the comment is not necessary and just confusing.

Reply:
I'm going to re-test with custom images containing patched anaconda package builds to verify whether the property is really needed. If it is, I'll tweak the comment somehow.

Reply:
So I tested, and it really does seem to need to be a property (or I guess maybe an instance variable? didn't test that yet). It just doesn't work if it's a class variable.

I'll twiddle the comment a bit I guess.